plugins/browser-plugin-youtube-tracking/src/index.ts

```diff
@@ -243,6 +243,8 @@ function youtubeEvent(player: YT.Player, eventName: string, conf: TrackingOption
   }
 
   const event = buildYouTubeEvent(player, eventName, conf, eventData);
+  /* FIXME: Convert CommonEventProperties interface to accept generics for SelfDescribingJson */
+  // @ts-expect-error
```

Comment:
Can we not add an index signature to `interface YouTube` (https://github.com/snowplow/snowplow-javascript-tracker/blob/0dbff387084ea714b75c770dbd8fc0781b6d6981/plugins/browser-plugin-youtube-tracking/src/contexts.ts#L95) so that is can assigned to `Record<string, unknown>`? Seems nicer to me than ignoring the TS error and it matches the other interfaces like `interface SnowplowMediaPlayer` (https://github.com/snowplow/snowplow-javascript-tracker/blob/0dbff387084ea714b75c770dbd8fc0781b6d6981/plugins/browser-plugin-youtube-tracking/src/types.ts#L53).

You could even add `[key: string]: string | string[] | boolean | number | number[] | null | undefined;` if you really wanted to 😂 but I'd be tempted to just go for `[key: string]: unknown;`

Reply:
The thing is that `[key: string]: unknown;` is kneecapping TypeScript and imo it should be used mostly in cases where our library provides APIs on the clients with uknown requirements. I cannot see the benefit to do this on our interfaces, especially the ones with standard properties. 

The kneecap is visible just by the fact that we were missing an attribute and TS did not even complain. That is `avaliableQualityLevels` which is fixed in this PR.

The good practise here is mentioned on the FIXME comment above the error expectation. The only reason I chose not to do the fix in this PR is it would make it a bit large and out of scope for the fix.

WDYT ? _hope I understood your comment_

Reply:
I find TODOs and FIXMEs a little triggering, and would prefer to see it fixed properly if we need to. I appreciate adding the unknown index type isn't a wonderful solution. I'd be a bit fan of adding generics in (in a non breaking change if possible) and I don't mind a little scope creep as its starting to feel like the right time to solve this. The more new plugins we add that track specific entities (ecom, consent, etc) the more we'll likely need this.

Reply:
Will add in a PR for v4. Removing the change here.